base/compiler/ssair/inlining.jl

```diff
@@ -1722,32 +1736,35 @@ function late_inline_special_case!(
     return nothing
 end
 
-function ssa_substitute!(idx::Int, @nospecialize(val), arg_replacements::Vector{Any},
+function ssa_substitute!(insert_node_before!,
```

Comment:
This appear it will end up typed as `::Function`, and introduce a dynamic dispatch in this function. Will that be an issue?

Reply:
Yeah, it should probably be specialized. Is there a way to annotate that? I guess alternatively, I could just create a small struct for it.